bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/ConfigTest.java

```diff
@@ -131,6 +133,106 @@ public void testTomlConfigProviderWithArrays() {
         Assert.assertEquals(((BArray) configValueMap.get(keys[5])).getValues(), expectedDecimalArray);
     }
 
+    @Test
+    public void testTomlProviderWithString() {
```

Comment:
We can revisit the unit tests in [#29381](https://github.com/ballerina-platform/ballerina-lang/pull/29381) to migrate negative tests separately and use data providers